.github/workflows/ci-rs.yml

```diff
@@ -347,6 +357,8 @@ jobs:
         uses: KyleMayes/install-llvm-action@v2
         with:
           version: ${{ env.LLVM_VERSION }}
+      - name: Install z3
```

Comment:
Driveby z3 dependency? If this is new, it'd be good to mention in the PR description perhaps.

LGTM otherwise.